libfoundation/src/foundation-string.cpp

```diff
@@ -4983,9 +4987,60 @@ bool MCStringConvertToSysString(MCStringRef p_string, const char * &r_system_str
     t_sys_string[t_sys_len] = '\0';
 
 	r_system_string = t_sys_string;
+    r_byte_count = t_sys_len;
+    
 	return true;
 }
 
+#elif defined(__MAC__)
+bool MCStringConvertToSysString(MCStringRef p_string, char *& r_system_string, size_t& r_byte_count)
+{
+    bool t_success;
+    uindex_t t_byte_count;
+    if (!MCStringConvertToUTF8(p_string, r_system_string, t_byte_count))
+        return false;
```

Comment:
That's the same as for MCStringConvertToSysString for Linux, the NIL byte is not included in the count returned by MCUnicodeCharsmapToUTF8.
I don't see any issue in excluding the nil byte from the count on Linux and Mac, but it is included in the length returned by WideCharToMultiBytes.
That said, we can certainly leave it as it is.